GameOfLife/GameOfLife/Utils/Panel.cs

```diff
@@ -0,0 +1,163 @@
+        {
+            Console.WriteLine($"Welcome to the {gameName}!");
+        }
+
+        /// <summary>
+        /// Arranges statistics in table-like appearance.
+        /// </summary>
+        /// <param name="stats">Statistics from the game.</param>
+        public static void DisplayStatsTable(Dictionary<string, int> stats)
+        {
+            foreach (var stat in stats)
+            {
+                Console.Write("{0, -15} {1,5}\n", stat.Key, stat.Value);
+            }
+
+            Console.WriteLine();
+        }
+
+        /// <summary>
+        /// Displays arrow-controlled menu with options from which player can choose.
+        /// </summary>
+        /// <param name="options">Options displayed with action.</param>
+        public static int DisplayOptionMenu(string[] options, string message)
+        {
+            int chosenOption = 0;
+
+            ConsoleKey key;
+
+            do
+            {
+                Console.ForegroundColor = ConsoleColor.White;
+                DisplayMessage(message);
+                
+
+                for (int i = 0; i < options.Length; i++)
+                {
+                    if (i == chosenOption)
+                    {
+                        Console.ForegroundColor = ConsoleColor.Green;
+                        Console.WriteLine(options[i]);
+                    }
+                    else
+                    {
+                        Console.ForegroundColor = ConsoleColor.White;
+                        Console.WriteLine(options[i]);
+                    }
+                }
+
+                key = Console.ReadKey().Key;
+
+                switch(key)
+                {
+                    case ConsoleKey.UpArrow:
+                        chosenOption--;
+
+                        if (chosenOption == -1)
+                        {
+                            chosenOption = options.Length - 1;
+                        }
+                        break;
+                        
+                    case ConsoleKey.DownArrow:
+                        chosenOption++;
+
+                        if (chosenOption == options.Length)
+                        {
+                            chosenOption = 0;
+                        }
+                        break;
+                }
+
+                Console.Clear();
+            }
+            while (key != ConsoleKey.Enter);
+
+            Console.ForegroundColor = ConsoleColor.White;
+
+            return chosenOption;
+        }
+
+        /// <summary>
+        /// Calls function on key press.
+        /// Programmable key fucntionality.
+        /// </summary>
+        /// <param name="spacebar">Action invoked by spacebar key.</param>
+        /// <param name="enter">Action invoked b  by escape key.</param>
+        /// <param name="s">Action inved by s key.</param>
+        public static void DisplayKeyMenu(Action spacebar, Action enter, Action escape, Action s)
```

Comment:
It is great you know and can use predefined delegates, but this implementation is too complex and not appropriate for this method.
This method is:  DisplayKeyMenu, so it should just display menu. 
For retrieving users input there should be other method and probably there should be master method using those methods.

Reply:
Solution simplified. Delegate thing removed.